### DISCUSSION TITLE
getting error while configuring a new Authentication Filter of keycloak on geoserver (trying to integrate keycloak SSO with geoserver)

### DISCUSSION THREAD

**Discussion by prakshe-verdethos** [keycloak/keycloak#20435] [Q&A]
![0GxEo](https://github.com/keycloak/keycloak/assets/126150087/17b0fc1c-11ae-4834-a83d-e62258183a72)
getting below error while integrating geoserver with keycloak

(

org.geoserver.security.validation.FilterConfigException: java.lang.RuntimeException: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "verify-token-audience" (class org.keycloak.representations.adapters.config.AdapterConfig), not marked as ignorable (40 known properties: "ssl-required", "cors-allowed-headers", "register-node-period", "redirect-rewrite-rules", "turn-off-change-session-id-on-login", "truststore", "client-key-password", "token-store", "resource", "realm", "proxy-url", "disable-trust-manager", "bearer-only", "autodetect-bearer-only", "truststore-password", "use-resource-role-mappings", "client-keystore", "confidential-port", "enable-pkce", "auth-server-url", "cors-allowed-methods", "public-client", "expose-token", "token-minimum-time-to-live", "min-time-between-jwks-requests", "ignore-oauth-query-parameter", "cors-max-age", "enable-cors", "allow-any-hostname", "realm-public-key", "credentials", "principal-attribute", "always-refresh-token", "policy-enforcer", "public-key-cache-ttl", "connection-pool-size", "client-keystore-password", "enable-basic-auth", "cors-exposed-headers", "register-node-at-startup"\]) at \[Source: (ByteArrayInputStream); line: 6, column: 32\] (through reference chain: org.keycloak.representations.adapters.config.AdapterConfig\["verify-token-audience"\])

)

Geoserver (2.23), keycloak (2.21)

**Comment by Anan0201** [comment]
I have the same probleme
were you able to solve it?

**Comment by danielFesenmeyer** [comment]
It seems like geoserver is not compatible with your keycloak version - which is probably NOT 2.21, because current keycloak version is **22.0.1**.
The geoserver developers should probably better ignore fields they don't understand, instead of throwing an error. ("Tolerant reader principle", will prevent several of such incompatibilities in the future.)
Maybe you can open a ticket there.
